content/articles/introduction-to-gui-using-tkinter-in-python/index.md

```diff
@@ -0,0 +1,162 @@
+
+Online hacking is growing more common these days, resulting to passwords and login information being compromised on a regular basis. If your passwords fall into the wrong hands, it can be highly inconvenient, and can even result to the loss of money or sensitive personal information.
+
+Unfortunately, the majority of stolen passwords are simple and easy to guess. Names, birthdays, hobbies, and favorite pets don't make good passwords, and using them is a recipe for disaster. Password generators come very handy in this situation.
+
+For personal or corporate usage, the best password generators make it simple to create strong passwords that are difficult to guess or crack.
+
+Let us see how we can build a password generator.
+
+Let's start by importing Tkinter.
+
+```Python
+from tkinter import *
+from tkinter.ttk import Combobox
+import random
+```
+
+The `tkinter.ttk` module gives the user access to the Tkinter widget set and allows the user to import `Combobox`. A combobox is a graphical user interface component that combines a drop-down box, list box, or allows the user to select input.
+
+```Python
+screen = Tk()
+screen.title("Password Generator")
+screen.geometry('600x400')
+screen.configure(background ="bisque")
+``` 
+
+A root window is created using the `Tk` class. The `Tk()` function assists in the creation of this GUI window, and provides numerous options such as setting the title and the geometry of the GUI window. The `geometry()` method is one of the many methods provided by Tkinter. We'll be working on the backdrop color of the Tkinter GUI window in this line of code.
+
+```Python
+def gen():
+   global sc1
+   sc1.set("")
+   passw=""
+   length=int(c1.get())
+   lowercase='abcdefghijklmnopqrstuvwxyz'
+   uppercase='ABCDEFGHIJKLMNOPQRSTUVWXYZ'+lowercase
+   mixs='0123456789'+lowercase+uppercase+'@#$%&*'
+```
+
+The global variable `sc1` is defined in the line code `global sc1`. We used the `.set()` method to set the value of `sc1` to null. We also assign a null value to the `passw` variable. The `length` variable will retain the value of `c1`, and the value type of `c1` will be integer. We then create a lowercase variable that holds a string value. After that, we create the `uppercase` variable to hold a string value. And at the last line of code, we define the `mixs`.
+
+Add the following code.
+
+```Python
+if c2.get()=='Low Strength':
+    for i in range(0,length):
+        passw=passw+random.choice(lowercase)
+    sc1.set(passw)
+elif c2.get()=='Medium Strength':
+    for i in range(0,length):
+        passw=passw+random.choice(uppercase)
+    sc1.set(passw)
+elif c2.get()=='High Strength':
+    for i in range(0,length):
+        passw=passw+random.choice(mixs)
+    sc1.set(passw)
+```
+
+We define the `if` condition to monitor the strength of the passwords generated using the password generator.
```

Comment:
Explain the rules used to determine the password strength.

Reply:
@geoffrey45  Done.
Thank you